### PR TITLE
Split Dag Generation and actual mining

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
This PR splits the dag generation and the mining algorithm.
This should make it easier to reuse the dag generation for progpow later on.

### CODE REVIEW

File: libethash-cuda/CMakeLists.txt [hunk 1]
```diff
@@ -28,8 +28,8 @@ else()
 	endif()
 endif()
 
-file(GLOB sources "*.cpp" "*.cu")
-file(GLOB headers "*.h" "*.cuh")
+file(GLOB sources "dag_generation_kernel.cu" "ethash/*.cu" "ethash/*.cpp")
```
Comment (chfast) [inline libethash-cuda/CMakeLists.txt]: Just list all the files.

Comment (MariusVanDerWijden) [reply libethash-cuda/CMakeLists.txt]: done

**Comment by AndreaLanfranchi** [comment]
@MariusVanDerWijden can you please rebase on this branch ?
https://github.com/AndreaLanfranchi/ethminer/tree/ProgPoW

File: libethash-cuda/CMakeLists.txt [hunk 2]
```diff
@@ -28,8 +28,8 @@ else()
 	endif()
 endif()
 
-file(GLOB sources "*.cpp" "*.cu")
-file(GLOB headers "*.h" "*.cuh")
+file(GLOB sources "dag_generation_kernel.cu" "ethash/ethash_cuda_miner_kernel.cu" "ethash/CUDAMiner.cpp")
```
Comment (chfast) [inline libethash-cuda/CMakeLists.txt]: But without `GLOB` :)

```suggestion
set(sources
    dag_generation_kernel.cu
    ethash/ethash_cuda_miner_kernel.cu
    ethash/CUDAMiner.cpp
    CUDAMiner.h
    ethash/ethash_cuda_miner_kernel.h
)
```

Comment (MariusVanDerWijden) [reply libethash-cuda/CMakeLists.txt]: Will change on other branch :)

**Comment by MariusVanDerWijden** [comment]
Yes, rebased it

**Comment by AndreaLanfranchi** [comment]
There were also changes on vectorize

**Comment by MariusVanDerWijden** [comment]
and the __device__ __forceinline__ macro.

**Pull request closed without merging**